components/script/script_thread.rs

```diff
@@ -3386,13 +3386,15 @@ impl ScriptThread {
     /// argument until a notification is received that the fetch is complete.
     fn pre_page_load(&self, mut incomplete: InProgressLoad, load_data: LoadData) {
         let id = incomplete.pipeline_id;
+        let top_level_browsing_context_id = incomplete.top_level_browsing_context_id;
         let req_init = RequestBuilder::new(load_data.url.clone(), load_data.referrer)
             .method(load_data.method)
             .destination(Destination::Document)
             .mode(RequestMode::Navigate)
             .credentials_mode(CredentialsMode::Include)
             .use_url_credentials(true)
             .pipeline_id(Some(id))
+            .target_browsing_context_id(Some(top_level_browsing_context_id))
```

Comment:
I moved the `target_browsing_context_id` setting here. It should be more appropriate.